pandas/tseries/offsets.py

```diff
@@ -2145,8 +2166,41 @@ def onOffset(self, dt):
 
 def _tick_comp(op):
     def f(self, other):
-        return op(self.delta, other.delta)
+        if isinstance(other, Tick):
+            # Note we cannot just try/except other.delta because Tick.delta
+            # returns a Timedelta while Timedelta.delta returns an int
+            other_delta = other.delta
+        elif isinstance(other, (timedelta, np.timedelta64)):
+            other_delta = other
+        elif isinstance(other, Week) and other.weekday is None:
+            other_delta = timedelta(weeks=other.n)
+        elif isinstance(other, compat.string_types):
+            from pandas.tseries.frequencies import to_offset
+            other = to_offset(other)
+            if isinstance(other, DateOffset):
+                return f(self, other)
+            else:
+                if op == operator.eq:
```

Comment:
why are you repeating

isn't
``return op(self, other)`` enough here?

Reply:
You mean in line 2181 returning `f(self, other)` instead of `op(self, other)`?  Because the the behavior still depends on which DateOffset subclass `other` is.